### ISSUE TITLE
Composition does not update addon tabs across refs

### ISSUE THREAD

**Issue by inthegreenwild** [storybookjs/storybook#12468]
**Describe the bug**
Addons do not update properly when switching between referenced storybooks. When I switch to "Three" I should have access to knobs and accessibility addons but I don't. I also still see "controls" which three package does not have.

**To Reproduce**

Clone the repo https://github.com/inthegreenwild/storybook-bug

Run commands: 

npm i
npm run prepare 
npm run preview

Open browser: 
http://localhost:6001

**Expected behavior**
As I move across storybooks, the addons tab should reflect what the storybook collection uses. 

**Screenshots**
![Screen Shot 2020-09-13 at 9 26 45 PM](https://user-images.githubusercontent.com/8966510/93037410-fbec6900-f607-11ea-8faa-b04bbabd2c2f.png)

![Screen Shot 2020-09-13 at 9 26 21 PM](https://user-images.githubusercontent.com/8966510/93037387-f000a700-f607-11ea-8e9c-8c8e57131fd0.png)

**System:**
  System:
    OS: macOS 10.15.6
    CPU: (6) x64 Intel(R) Core(TM) i5-8500B CPU @ 3.00GHz
  Binaries:
    Node: 14.10.1 - ~/.nvm/versions/node/v14.10.1/bin/node
    Yarn: 1.22.5 - ~/.nvm/versions/node/v14.10.1/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.10.1/bin/npm
  Browsers:
    Chrome: 85.0.4183.102
    Firefox: 80.0
    Safari: 13.1.2

**Additional context**

**Comment by shilman** [comment]
Closing as dupe to #12278 (tho i realize it's not exactly identical)
